infrastructure/subscribers/src/main/java/tech/pegasys/teku/infrastructure/subscribers/Subscribers.java

```diff
@@ -49,12 +50,21 @@
 
   private final boolean suppressCallbackExceptions;
 
-  private Subscribers(final boolean suppressCallbackExceptions) {
+  private final boolean replayLastActionOnSubscription;
+  private Optional<Consumer<T>> lastAction = Optional.empty();
+
+  private Subscribers(
+      final boolean suppressCallbackExceptions, final boolean replayLastActionOnSubscription) {
     this.suppressCallbackExceptions = suppressCallbackExceptions;
+    this.replayLastActionOnSubscription = replayLastActionOnSubscription;
   }
 
   public static <T> Subscribers<T> create(final boolean suppressCallbackExceptions) {
-    return new Subscribers<>(suppressCallbackExceptions);
+    return new Subscribers<>(suppressCallbackExceptions, false);
+  }
+
+  public static <T> Subscribers<T> createWithReplay(final boolean suppressCallbackExceptions) {
+    return new Subscribers<>(suppressCallbackExceptions, true);
```

Comment:
I'm quite sceptical about adding this here.  The need to add synchronisation has a big impact across a lot of our code base.  I suspect it could introduce some potential deadlocks as well if events are ever raised while holding some other lock.

We do have the `ObservableValue` class which is specifically designed to manage a single value and notify of updates, including providing the current value to new subscribers.

Reply:
I agree with you, but there is a case in async environment when we could not guarantee the order, I've described it above, may be we could come to better solution for it

Reply:
Sounds like we should, probably in a PR by itself, move from using `Subscriber` to `ObservableValue` then.

Reply:
Good, I will take a look at it